jetpack.php

```diff
@@ -93,6 +93,21 @@ function jetpack_admin_unsupported_wp_notice() { ?>
  */
 $jetpack_autoloader = JETPACK__PLUGIN_DIR . 'vendor/autoload_packages.php';
 if ( is_readable( $jetpack_autoloader ) ) {
+	add_filter( 'upgrader_post_install', 'jetpack_refresh_autoloader_post_install', 1, 2 );
+	function jetpack_refresh_autoloader_post_install( $worked, $hook_extras ) {
+		error_log( print_r( $hook_extras, 1 ) );
+		if (
+			! isset( $hook_extras['plugin'] )
+			|| JETPACK__PLUGIN_FILE !== $hook_extras['plugin']
+		) {
+			return $worked;
+		}
+		// Include the new autoloader since the classes might have been shifter.
+		include JETPACK__PLUGIN_DIR . 'vendor/autoload_packages.php';
+
+		return $worked;
+	}
+
 	require $jetpack_autoloader;
```

Comment:
Since we may include it above, should we change this to `require_once`?

Reply:
I think `require` is ok here. This file, `jetpack.php`, should only be loaded once, so this `require` is only called once. The `include` statement in line 112 is in the `jetpack_refresh_autoloader_post_install()` function, which is hooked to `upgrader_post_install`. It will be called after the `require`.